python/ray/workflow/tests/test_wait.py

```diff
@@ -0,0 +1,150 @@
+from ray.tests.conftest import *  # noqa
```

Comment:
The tests are a little hard to read with the workflow steps interleaved with the test code. Can you move the workflow steps into the test functions? It's fine if we have to duplicate them.

Also, can you add comments to explain what logic you're testing for?